addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js

```diff
@@ -133,6 +134,12 @@ class Sanitize {
             }
         }
 
+        if (isONotEditable(node)) {
+            if (node.getAttribute('contenteditable') !== 'false') {
+                node.setAttribute('contenteditable', 'false');
+            }
+        }
```

Comment:
The problem in question is actually caused by commit 1b80fe9b56acea09a6784691561df65fe6349f50. Before this commit, the `contenteditable` attribute was added by the same `MutationObserver` that adds the `o_dirty` classes. The fix does not actually depend on the removal of this code, so it was probably removed by mistake. Even now the explanatory comment is still there (`1. Make sure every .o_not_editable is not editable.`), however the code has disappeared.

Restoring the old behavior also fixes the problem (see commit 10eba7d5d964dffa49a74085b05de8ae831fa15a, I also stumbled on this bug) however, adding the fix to `Sanitize` seems to make more sense. However, the fix should probably also include:
- The removal of the comment
- The removal of the `contenteditable` attribute check in the `MutationObserver` mentioned above, since these mutations are already being filtered out by `filterMutationRecords`, as they should.

Reply:
Indeed, the old behavior seems to have been mistakenly removed.
You're commit seems on the right direction, however:

1) Although there is already specific code to Odoo mixed with the editor lib, we try to avoid it as much as possible.
Therefore, `isONotEditable` in `lib/odoo-editor/src/utils/sanitize.js` should be avoided.

2) The behavior will work only if a mutation observer is active when the `o_not_editable` is called.

I've made a commit that fixes both issue on that branch: https://github.com/odoo-dev/odoo/tree/15.0-not-editable-o_not_editable-nby

As @dmo-odoo is currently on vacation until next week, you can cherry pick the commit if merging this PR is urgent.

Reply:
@Goaman This PR is not "urgent" but the fix (related to multiple PR and simply on its own) is. So if you can create a PR for your branch so I can have a look and merge, that would be nice :+1: Also what about the other system merged by your team (I mentioned in https://github.com/odoo/odoo/pull/82222#discussion_r829173375)? Should it be kept alongside that o_not_editable behavior (I did not check in detail how it works)?

We can continue the discussion on the PR you will create, I guess it will be better :+1: Thanks :+1:

Reply:
Allright. PR: https://github.com/odoo/odoo/pull/87347